public/components/TagEdit/formComponents/TagImageEdit.react.js

```diff
@@ -0,0 +1,152 @@
+import React from 'react';
+import {validateImageUrl} from '../../../util/validateImage';
+import {getImageMetadata} from '../../../util/supportApi.js';
+
+const FETCH_STATES = {
+  error: 'FETCH_STATE_ERROR',
+  fetching: 'FETCH_STATE_FETCHING',
+  success: 'FETCH_STATE_SUCCESS'
+};
+
+export default class TagImageEdit extends React.Component {
+
+  constructor(props) {
+    super(props);
+
+    this.state = {
+      inputUrl: ''
+    };
+
+    this.fetchMetadata = this.fetchMetadata.bind(this);
+  }
+
+  onImageUpdate() {
+    this.props.onChange();
+  }
+
+  getMainAsset() {
+
+    if (!this.props.tagImage.assets) {
+      return false;
+    }
+
+    return this.props.tagImage.assets.sort((a, b) => a.width < b.width)[0];
+  }
+
+  removeImage() {
+    this.props.onChange({});
+  }
+
+  updateInputUrl(e) {
+    this.setState({
+      inputUrl: e.target.value,
+      currentMetadata: false,
+      metadataFetchStatus: FETCH_STATES.fetching
+    });
+
+    this.fetchMetadata(e.target.value);
+  }
+
+  addImage() {
+    this.props.onChange({
+      imageId: this.state.currentMetadata.id,
+      assets: [
+        {
+          height: this.state.currentMetadata.height,
+          imageUrl: this.state.inputUrl,
+          mimeType: this.state.currentMetadata.mimeType,
+          width: this.state.currentMetadata.width
+        }
+      ]
+    });
+
+    this.setState({
+      inputUrl: undefined,
+      currentMetadata: undefined,
+      metadataFetchStatus: undefined
+    });
+  }
+
+  fetchMetadata(imageUrl) {
+
+    //Validate before any requests happen
+    if (!validateImageUrl(imageUrl)) {
+      this.setState({
+        currentMetadata: false,
+        metadataFetchStatus: FETCH_STATES.error
+      });
+
+      return false;
+    }
+
+    getImageMetadata(imageUrl).then(metadata => {
+      if (this.state.inputUrl === imageUrl) { // Check it's still the current url
+        this.setState({
+          currentMetadata: metadata,
+          metadataFetchStatus: FETCH_STATES.success
+        });
+      }
+    }).fail(e => {
+      this.setState({
+        currentMetadata: false,
+        metadataFetchStatus: FETCH_STATES.error
+      });
+    });
+  }
+
+  renderAddButton() {
+    if (this.state.metadataFetchStatus === FETCH_STATES.fetching) {
+      return (<div className="tag-edit__image__add">Checking Url...</div>);
+    }
+
+    if (this.state.metadataFetchStatus === FETCH_STATES.error) {
+      return (<div className="tag-edit__image__add--error"><i className="i-cross-red" />Url Not Valid</div>);
```

Comment:
Maybe add a bit more explanation about what constitutes a valid image url, that it needs to be on static.guim... etc.